src/adagrams.js

```diff
@@ -1,15 +1,178 @@
+
+
+export const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+
+
+
+/*********** Helper Functions ****************/
+export let letterArr = () => {
+  let allLetters = [];
+  for (const [letter, freq] of Object.entries(LETTER_POOL)){
+    let currentLetterArr = new Array(freq).fill(letter);
+    // Spread syntax ...
+    allLetters.push(...currentLetterArr);
+    
+  }
+  return allLetters;
+};
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  let currentLetterBank = letterArr();
+  const deck = [];
+
+  // SHUFFLE THE LIST OF ALL LETTERS - AKA shuffle the current letter bank
+  // Fisher-Yates Shuffle Algorithm
+  // Reference: https://www.tutorialspoint.com/what-is-fisher-yates-shuffle-in-javascript
+  for (let index = currentLetterBank.length-1; index > 0; index--){
+    const randomIndex = Math.floor(Math.random() * (index+1));
+    
+    const temp = currentLetterBank[index];
+    currentLetterBank[index] = currentLetterBank[randomIndex];
+    currentLetterBank[randomIndex] = temp;
+  }
+
+
+  while (deck.length < 10) {
+    // Randomly select an index from the shuffled list of all letters
+    const randoIdx = Math.floor(Math.random() * currentLetterBank.length);
+    const randoLetter = currentLetterBank[randoIdx];
+    deck.push(randoLetter);
+    currentLetterBank.splice(randoIdx, 1);
+  }
+
+  return deck;
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  let newList = [...lettersInHand];
+  const lastLetter = input.length-1;
+  const letterCheck = newList.includes(input[lastLetter]);
+  const idx = lettersInHand.indexOf(input[lastLetter]);
+
+  /* Base Cases */
+  if (!letterCheck) {
+    return false;
+  }
+
+  else if (letterCheck && input.length == 1) {
+    newList.splice(idx, 1);
+    return true;
+  }
+
+  /* Recursive Case */
+  else if (letterCheck && input.length != 1) {
+    newList.splice(idx, 1);
+    return usesAvailableLetters(input.slice(0, lastLetter), newList);
+  }
```

Comment:
Love the recursion practice, just keep a couple of things in mind about it. Though recursion can come in handy, there are a lot of times where it puts our code at a disadvantage technically (time complexity) and also readability. Recursion tends to work best on mirror data structures, think a directory tree.